### PR TITLE
CVE-2015-1252, CVE-2015-1361

### PR DESCRIPTION
**Pull request by ZacharyMoran**
Updated Research Docs to Include My Findings on the two CVEs

### CODE REVIEW

File: cves/CVE-2015-1252.yml [hunk 1]
```diff
@@ -124,7 +145,19 @@ interesting_commits:
...
+    an out of bounds write that occured after the initial portion of the data was written to
+    the buffer. The only commits that occured between the VCC (initial implementation) and 
+    the fix was one to add append support to the PartialCircularBuffer instead of just overwrite
+    capabilities. Originally this was of interest but after further investigation it was determined
+    that the problem was with the core implementation of the DoWrite and Write functionality and
+    the additional support for append did not affect it at all as it was simply using the same code
+    with an additional switch to determine whether or not to clear the buffer prior to attempting a
+    write. This append support was to be used in logging machine info to diagnostic the WebRTC log.
+    The only other commit that occured was moving the file from content/common to chrome/common and
+    changing the copyright info.
```
Comment (kbaumzie) [inline cves/CVE-2015-1252.yml]: I appreciate how thorough your research is!

File: cves/CVE-2015-1252.yml [hunk 2]
```diff
@@ -205,6 +238,17 @@ mistakes:
...
+    that is the exact issue that occured but the NVD listed CWE-119 has what
+    I would consider applicable, specifically the mitigations in the
+    implementation stage. The fix is specifically addressing the mitigation
+    of checking the buffer boundaries if accessing the buffer in a loop
+    to make sure you are not in danger of writing past the allocated space.
+    Since the issue at its core is a lack of checking proper space checking
+    on the buffer size vs input size when handling the wrap around case, the
+    fix moves the wrap around content out of the initial write loop to ensure
+    safety when writing the given input. The initial implementation simply did
+    not include thorough enough testing to catch this edge case.
```
Comment (bjpicciano) [inline cves/CVE-2015-1252.yml]: This is good to note. You may think your tests cover all cases and writing new ones would be redundant, but in this case the right case would have caught this issue pretty easily.

File: cves/CVE-2015-1361.yml [hunk 3]
```diff
@@ -4,14 +4,14 @@ CWE_instructions: |
   Please go to cwe.mitre.org and find the most specific, appropriate CWE entry
   that describes your vulnerability. (Tip: this may not be a good one to start
   with - spend time understanding this vulnerability before making your choice!)
-CWE: 
+CWE: 17
```
Comment (bjpicciano) [inline cves/CVE-2015-1361.yml]: Hmm, I'm not sure what CWE-17 is about. Is it just a generic coding error? It feel weird, but if it has no effect on the system, I guess it fits well!

Comment (ZacharyMoran) [reply cves/CVE-2015-1361.yml]: Yeah I was not entirely sure what to put here as the vulnerability could have been a bunch of different things if it actually affected something, but since it didn't I assumed this was sufficient.

File: cves/CVE-2015-1252.yml (hunk 1 above)
Comment (ZacharyMoran) [reply cves/CVE-2015-1252.yml]: Thank you!

**Comment by sxHelloWorld** [comment]
Overall, your researches look excellent. I give +4 to both CVE (8 total).

File: cves/CVE-2015-1252.yml [hunk 4]
```diff
@@ -11,7 +11,7 @@ curated_instructions: |
   integrity checks on this file to make sure you fill everything out properly.
   If you are a student, we cannot accept your work as finished unless curated is
   set to true.
-curated: false
+curated: true
```
Comment (bjpicciano) [inline cves/CVE-2015-1252.yml]: +3 upvotes

File: cves/CVE-2015-1361.yml (hunk 3 above)
Comment (bjpicciano) [inline cves/CVE-2015-1361.yml]: +1 upvote
